workspace-loader/src/workspace-loader.ts

```diff
@@ -26,6 +26,10 @@ export class WorkspaceLoader {
     private workspace: che.workspace.Workspace;
     private runtimeIsAccessible: Deferred<void>;
 
+    // `false` if workspace has been stopped intentionally
+    // and workspace-loader should not restart it
+    private doRestart: boolean = true;
```

Comment:
I'm confused. Restart means the sequence of actions: stop and then start. Is this variable indeed means that?

Reply:
It's not the best name, indeed. I suppose that `allowRestart` makes more sense, doesn't it?

Reply:
Let it be `allowRestart`